src/Mvc/Mvc/src/MvcServiceCollectionExtensions.cs

```diff
@@ -88,13 +88,7 @@ public static IMvcBuilder AddMvc(this IServiceCollection services, Action<MvcOpt
     /// </remarks>
     public static IMvcBuilder AddControllers(this IServiceCollection services)
     {
-        if (services == null)
-        {
-            throw new ArgumentNullException(nameof(services));
-        }
-
-        var builder = AddControllersCore(services);
-        return new MvcBuilder(builder.Services, builder.PartManager);
+        services.AddControllers(null);
```

Comment:
@arialdomartini Thanks for this PR. Probably you're still working on it, since there are some build errors but I would prefer to change the `AddControllers(this IServiceCollection services, Action<MvcOptions>? configure)` instead to call the `AddControllers(this IServiceCollection services)` that is a more common pattern in the repo. Something like this:

``` c#
public static IMvcBuilder AddControllers(this IServiceCollection services, Action<MvcOptions>? configure)
{
        // This method excludes all of the view-related services by default.
        var builder = services.AddControllers();
        if (configure != null)
        {
            builder.Services.AddMvcOptions(configure);
        }

        return builder;
}
```

Reply:
Eg.: https://github.com/dotnet/aspnetcore/blob/797623cd2125901019b9f3336da80d4b934cc6f7/src/Mvc/Mvc/src/MvcServiceCollectionExtensions.cs#L46